### PR TITLE
Add Cluster Metrics instructions

### LINKED ISSUE #438: [FEATURE] ClusterMetrics monitor
A new monitor type, ClusterMetricsMonitor, will allow users to create monitors that alert users of information returned by `GET _cat`. 

For 1.3, add this new monitoring type somewhere in the [Monitors section](https://opensearch.org/docs/latest/monitoring-plugins/alerting/monitors/).

## Resources

See Design documented provided by Thomas Hurney.

### PR DESCRIPTION
**Pull request by Naarcha-AWS**
Signed-off-by: Naarcha-AWS <naarcha@amazon.com>

### Description
Adds Cluster Metrics documentation
 
### Issues Resolved
Fixes #438 
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _monitoring-plugins/alerting/monitors.md [hunk 1]
```diff
@@ -406,3 +405,112 @@ Acknowledged | Someone has acknowledged the alert, but not fixed the root cause.
 Completed | The alert is no longer ongoing. Alerts enter this state after the corresponding trigger evaluates to false.
 Error | An error occurred while executing the trigger---usually the result of a a bad trigger or destination.
 Deleted | Someone deleted the monitor or trigger associated with this alert while the alert was ongoing.
+
+---
+
+## Create cluster metrics monitor
+
+In addition to monitoring conditions for indexes, the alerting plugin allows monitoring conditions on clusters. Alerts that can be set by ClusterMetrics to watch for when:
```
Comment (lizsnyder) [inline _monitoring-plugins/alerting/monitors.md:413]: Did you mean "alerts can be set"?

File: _monitoring-plugins/alerting/monitors.md [hunk 2]
```diff
@@ -406,3 +405,112 @@ Acknowledged | Someone has acknowledged the alert, but not fixed the root cause.
...
+- JVM memory usage reaches a specified threshold.
+- The total number of documents stores reaches a specified amount.
+
+To create a cluster metrics monitor:
+
+1. Select **Alerting** > **Monitors** > **Create monitor**.
+2. Select the **Per cluster metrics monitor** option. 
+3. In the Query section, pick the **Request type** from the dropdown.
+4. (Optional) If you want to filter the API response to use only certain path parameters, enter those parameters under **Query parameters**. Most APIs that can be used to monitor cluster status support path parameters as described in their documentation (e.g., comma-separated lists of index names).
+5. In the Triggers section, indicate what conditions trigger an alert. The trigger condition autopopulates a painless ctx variable. For example, a cluster monitor watching for Cluster Stats uses the trigger condition `ctx.results[0].indices.count <= 0`, which triggers an alert based on the number of indices returned by the query. For more specificity, add any additional painless conditions supported by the API. To see an example of the condition response, select **Preview condition response**.
```
Comment (lizsnyder) [inline _monitoring-plugins/alerting/monitors.md:426]: indexes

File: _monitoring-plugins/alerting/monitors.md [hunk 3]
```diff
@@ -406,3 +405,112 @@ Acknowledged | Someone has acknowledged the alert, but not fixed the root cause.
...
+
+1. Select **Alerting** > **Monitors** > **Create monitor**.
+2. Select the **Per cluster metrics monitor** option. 
+3. In the Query section, pick the **Request type** from the dropdown.
+4. (Optional) If you want to filter the API response to use only certain path parameters, enter those parameters under **Query parameters**. Most APIs that can be used to monitor cluster status support path parameters as described in their documentation (e.g., comma-separated lists of index names).
+5. In the Triggers section, indicate what conditions trigger an alert. The trigger condition autopopulates a painless ctx variable. For example, a cluster monitor watching for Cluster Stats uses the trigger condition `ctx.results[0].indices.count <= 0`, which triggers an alert based on the number of indices returned by the query. For more specificity, add any additional painless conditions supported by the API. To see an example of the condition response, select **Preview condition response**. 
+6. In the Actions sections, indicate how you want your users to be notified when a trigger condition is met.
+7. Select **Create**. Your new monitor appears in **Monitors** list.
```
Comment (lizsnyder) [inline _monitoring-plugins/alerting/monitors.md:428]: in the Monitors list

File: _monitoring-plugins/alerting/monitors.md [hunk 4]
```diff
@@ -406,3 +405,112 @@ Acknowledged | Someone has acknowledged the alert, but not fixed the root cause.
...
+3. [_cluster/settings](https://opensearch.org/docs/latest/opensearch/rest-api/cluster-settings/)
+4. [_nodes/stats](https://opensearch.org/docs/latest/opensearch/popular-api/#get-node-statistics)
+5. _[cat/pending_tasks](https://opensearch.org/docs/latest/opensearch/rest-api/cat/cat-pending-tasks/)
+6. [_cat/recovery](https://opensearch.org/docs/latest/opensearch/rest-api/cat/cat-recovery/
+7. [_cat/snapshots](https://opensearch.org/docs/latest/opensearch/rest-api/cat/cat-snapshots/)
+8. [_cat/tasks](https://opensearch.org/docs/latest/opensearch/rest-api/cat/cat-tasks/)
+
+### Restrict API fields
+
+If you want to hide fields from the API response that you do not want exposed for alerting, create a `supported_json_payloads.json` file in the alerting plugin. The file functions as an allow list for the API fields you want to use in an alert. Cluster metric monitors can only be created for API's referenced by the supported payloads file. Furthermore, only fields reference in the supported files can create trigger conditions.
```
Comment (lizsnyder) [inline _monitoring-plugins/alerting/monitors.md:445]: APIs (ditch apostrophe)

Also, "only fields reference**d**" I think?

File: _monitoring-plugins/alerting/monitors.md [hunk 5]
```diff
@@ -406,3 +405,112 @@ Acknowledged | Someone has acknowledged the alert, but not fixed the root cause.
...
+
+This ``supported_json_payloads.json` allows for a cluster metrics monitor to be created for the `_cluster/stats` API, and triggers conditions for the `indices.shards.total` and `indices.shards.index.shards.min` fields.
```
Comment (lizsnyder) [inline _monitoring-plugins/alerting/monitors.md:447]: I think you have an extra tick mark ``supported_json_payloads.json`

File: _monitoring-plugins/alerting/monitors.md [hunk 6]
```diff
@@ -406,3 +405,112 @@ Acknowledged | Someone has acknowledged the alert, but not fixed the root cause.
...
+     "indices": [
+         "shards.total",
+         "shards.index.shards.min"
+     ]
+}
+```
+
+### Painless triggers
+
+Painless scripts define triggers for cluster metrics monitors, similar to query or bucket-level monitors that are defined using the extraction query definition option. Painless scripts are compromised of at least one statement and any additional functions you wish to execute.
```
Comment (lizsnyder) [inline _monitoring-plugins/alerting/monitors.md:460]: comprised

File: _monitoring-plugins/alerting/monitors.md [hunk 7]
```diff
@@ -406,3 +405,112 @@ Acknowledged | Someone has acknowledged the alert, but not fixed the root cause.
...
+}
+```
+
+See [trigger variables](#trigger-variables), for more painless ctx options.
+
+### Limitations
+
+Currently, the cluster metrics monitor has the following limitations: 
+
+- You cannot create monitor for remote clusters.
```
Comment (lizsnyder) [inline _monitoring-plugins/alerting/monitors.md:513]: "create a monitor", or "create monitors"?

File: _monitoring-plugins/alerting/monitors.md [hunk 8]
```diff
@@ -406,3 +405,112 @@ Acknowledged | Someone has acknowledged the alert, but not fixed the root cause.
...
+3. [_cluster/settings](https://opensearch.org/docs/latest/opensearch/rest-api/cluster-settings/)
+4. [_nodes/stats](https://opensearch.org/docs/latest/opensearch/popular-api/#get-node-statistics)
+5. _[cat/pending_tasks](https://opensearch.org/docs/latest/opensearch/rest-api/cat/cat-pending-tasks/)
+6. [_cat/recovery](https://opensearch.org/docs/latest/opensearch/rest-api/cat/cat-recovery/
+7. [_cat/snapshots](https://opensearch.org/docs/latest/opensearch/rest-api/cat/cat-snapshots/)
+8. [_cat/tasks](https://opensearch.org/docs/latest/opensearch/rest-api/cat/cat-tasks/)
+
+### Restrict API fields
+
+If you want to hide fields from the API response that you do not want exposed for alerting, create a `supported_json_payloads.json` file in the alerting plugin. The file functions as an allow list for the API fields you want to use in an alert. Cluster metric monitors can only be created for APIs referenced by the supported payloads file. Furthermore, only fields referenced in the supported files can create trigger conditions.
```
Comment (AWSHurneyt) [inline _monitoring-plugins/alerting/monitors.md:445]: Perhaps reword this to "configure the `supported_json_payloads.json` file in the alerting plugin." The backend package contains this file already. It's located in the [opensearch-alerting/alerting/src/main/resources/org/opensearch/alerting/settings](https://github.com/opensearch-project/alerting/blob/main/alerting/src/main/resources/org/opensearch/alerting/settings/supported_json_payloads.json) directory. That file is currently configured to allow monitors to be created for all 8 of the supported API, and for the entire API response to be available for defining trigger conditions.

File: _monitoring-plugins/alerting/monitors.md [hunk 9]
```diff
@@ -406,3 +405,112 @@ Acknowledged | Someone has acknowledged the alert, but not fixed the root cause.
...
+
+To create a cluster metrics monitor:
+
+1. Select **Alerting** > **Monitors** > **Create monitor**.
+2. Select the **Per cluster metrics monitor** option. 
+3. In the Query section, pick the **Request type** from the dropdown.
+4. (Optional) If you want to filter the API response to use only certain path parameters, enter those parameters under **Query parameters**. Most APIs that can be used to monitor cluster status support path parameters as described in their documentation (e.g., comma-separated lists of index names).
+5. In the Triggers section, indicate what conditions trigger an alert. The trigger condition autopopulates a painless ctx variable. For example, a cluster monitor watching for Cluster Stats uses the trigger condition `ctx.results[0].indices.count <= 0`, which triggers an alert based on the number of indexes returned by the query. For more specificity, add any additional painless conditions supported by the API. To see an example of the condition response, select **Preview condition response**. 
+6. In the Actions sections, indicate how you want your users to be notified when a trigger condition is met.
```
Comment (AWSHurneyt) [inline _monitoring-plugins/alerting/monitors.md:427]: For the line "In the Actions sections," should "sections" be singular in this context?

Comment (Naarcha-AWS) [reply _monitoring-plugins/alerting/monitors.md:427]: Yes. Good catch.

**Review by lizsnyder — APPROVED** [review]

**Comment by Naarcha-AWS** [comment]
@AWSHurneyt: Let's merge this in for now. I'll add the new use case in a separate PR.
